docs: update client sdk get evaluation details (#155)
bucketeer-io/bucketeer-docs: docs/sdk/client-side/javascript/index.md

numberVariation(featureId: string, defaultValue: number): Promise<number>;

// The returned value will be either a BKTJsonObject or a BKTJsonArray. If no result is found, it will return the provided `defaultValue`, which can be of any type within `BKTValue`.

> objectVariation(featureId: string, defaultValue: BKTValue):Promise<object>;

Reviewer: ```suggestion
objectVariation(featureId: string, defaultValue: BKTValue):Promise<BKTValue>;
```

```

</TabItem>
</Tabs>

### Getting evaluation details

The following methods will return the **evaluation details** for a specific feature flag. If the feature flag is missing in the SDK's cache, the variable `reason` value will be `CLIENT`, which means the default value was returned.

This is useful if you use another A/B Test solution with Bucketeer and need to know the variation name, reason, and other information.

:::caution Deprecated

The `evaluationDetails` interface is deprecated. Please use the following [interfaces](#interfaces).

:::

#### Interfaces

<Tabs>
  <TabItem value="js" label="JavaScript">

```js showLineNumbers

> interface BKTClient {
>   booleanVariationDetails(
>     featureId: string,
>     defaultValue: boolean
>   ): BKTEvaluationDetails<boolean>;
> 
>   stringVariationDetails(
>     featureId: string,
>     defaultValue: string
>   ): BKTEvaluationDetails<string>;
> 
>   numberVariationDetails(
>     featureId: string,
>     defaultValue: number
>   ): BKTEvaluationDetails<number>;
> 
>   objectVariationDetails(
>     featureId: string,
>     defaultValue: BKTValue
>   ): BKTEvaluationDetails<BKTValue>;
> }

Reviewer: ```suggestion
interface BKTClient {
  booleanVariationDetails: (
    featureId: string,
    defaultValue: boolean
  ) => BKTEvaluationDetails<boolean>;

  stringVariationDetails: (
    featureId: string,
    defaultValue: string
  ) => BKTEvaluationDetails<string>;

  numberVariationDetails: (
    featureId: string,
    defaultValue: number
  ) => BKTEvaluationDetails<number>;

  objectVariationDetails: (
    featureId: string,
    defaultValue: BKTValue
  ) => BKTEvaluationDetails<BKTValue>;
}
```

  ```

  </TabItem>
</Tabs>

#### Object

<Tabs>
  <TabItem value="js" label="JavaScript">

```js showLineNumbers

> export interface BKTEvaluationDetails<T> {
>   featureId: string; // The ID of the feature flag.
>   featureVersion: number; // The version of the feature flag.
>   userId: string; // The ID of the user being evaluated.
>   variationId: string; // The ID of the assigned variation.
>   variationName: string; // The name of the assigned variation.
>   variationValue: T; // The value of the assigned variation.
>   reason: Reason; // The reason for the evaluation.
> }
> 
> export type Reason =
>   | "TARGET" // Evaluated using individual targeting.
>   | "RULE" // Evaluated using a custom rule.
>   | "DEFAULT" // Evaluated using the default strategy.
>   | "CLIENT" // The flag is missing in the cache; the default value was returned.
>   | "OFF_VARIATION" // Evaluated using the off variation.
>   | "PREREQUISITE"; // Evaluated using a prerequisite.

Reviewer: ```suggestion
export interface BKTEvaluationDetails<T extends BKTValue> {
  readonly featureId: string; // The ID of the feature flag.
  readonly featureVersion: number; // The version of the feature flag.
  readonly userId: string; // The ID of the user being evaluated.
  readonly variationId: string; // The ID of the assigned variation.
  readonly variationName: string; // The name of the assigned variation.
  readonly variationValue: T; // The value of the assigned variation.
  readonly reason: 
    | 'TARGET' // Evaluated using individual targeting.
    | 'RULE' // Evaluated using a custom rule.
    | 'DEFAULT' // Evaluated using the default strategy.
    | 'CLIENT' // The flag is missing in the cache; the default value was returned.
    | 'OFF_VARIATION' // Evaluated using the off variation.
    | 'PREREQUISITE'; // Evaluated using a prerequisite.
}
```
